src/nupic/algorithms/Connections.cpp

```diff
@@ -89,7 +80,7 @@ Synapse Connections::createSynapse(const Segment& segment,
                                    Permanence permanence)
 {
   vector<SynapseData>& synapses = cells_[segment.cell.idx].segments[segment.idx].synapses;
-  if (synapses.size() == SYNAPSE_MAX)
+  if (synapses.size() == MAX_SYNAPSES_PER_SEGMENT)
```

Comment:
I think it is very confusing as is. Fine if you want to do in a separate PR or in this one.